pkg/sql/catalog/replication/reader_catalog_test.go

```diff
@@ -409,6 +410,9 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 		for !iterationsDone {
 			uniqueIdx++
 			if !useAOST {
+				// Toggle the block on each iteration, so there is some risk
+				// of not all descriptor updates being updated.
+				descriptorRefreshHookEnabled.Swap(true)
```

Comment:
I'm finding it hard to reason through all the components of this test, perhaps because i'm unfamiliar with the LeaseManager hooks you're using. Perhaps in a follow up PR after we fix the bug, we could break this test up into a few smaller tests that are easier to understand.

Reply:
Fair point, and I think a bunch of code can be shared here too. I opened: #136114 for tracking
<!-- Reviewable comment -OCYqfGiA5HJ-v77CniN:b-eqox7r -->
<!-- Sent from Reviewable.io -->